sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/regexpExpressions.scala

```diff
@@ -162,10 +160,17 @@ case class Like(left: Expression, right: Expression, escapeChar: Char = '\\')
       } else {
         escapeChar
       }
+      val rightStr = ctx.freshName("rightStr")
+      val pattern = ctx.addMutableState(patternClass, "pattern")
+      val lastRightStr = ctx.addMutableState(classOf[String].getName, "lastRightStr")
+
       nullSafeCodeGen(ctx, ev, (eval1, eval2) => {
         s"""
           String $rightStr = $eval2.toString();
-          $patternClass $pattern = $patternClass.compile($escapeFunc($rightStr, '$newEscapeChar'));
+          if ($lastRightStr != $rightStr) {
```

Comment:
I think this optimization is a corner case and highly depends on input data.... cc: @dongjoon-hyun

Reply:
Yes, and here I have an another idea.
Now `BroadcastNestedLoopJoinExec` always loop streamed firstly then nest loop broadcast and the pattern is always at broadcast side.
In inner like join, we can add another optimization that loop broadcast first and then nest loop streamed. So we can use the feature of  this pr.